Source/santasyncservice/SNTPolaris.mm

```diff
@@ -0,0 +1,97 @@
+/// Copyright 2025 North Pole Security, Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    https://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+#include "Source/santasyncservice/SNTPolaris.h"
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#import <CommonCrypto/CommonCrypto.h>
+
+#include "Source/common/SNTLogging.h"
+#include "Source/common/SNTSystemInfo.h"
+
+#include <google/protobuf/arena.h>
+
+#include <grpc/grpc.h>
+#include <grpcpp/channel.h>
+#include <grpcpp/client_context.h>
+#include <grpcpp/create_channel.h>
+#include <grpcpp/security/credentials.h>
+
+#include "stats/v1.grpc.pb.h"
+
+namespace pbv1 = ::santa::stats::v1;
+
+namespace santa {
+
+const char *kPolarisHostname = "polaris.northpole.security";
+
+std::string machineIdHash(std::string machineID) {
+  CC_SHA256_CTX c256;
+  CC_SHA256_Init(&c256);
+  CC_SHA256_Update(&c256, machineID.data(), (CC_LONG)machineID.length());
+
+  unsigned char digest[CC_SHA256_DIGEST_LENGTH];
+  CC_SHA256_Final(digest, &c256);
+
+  std::stringstream ss;
+  for (size_t i = 0; i < CC_SHA256_DIGEST_LENGTH; i++) {
```

Comment:
Heh, I have a function I added in this open PR to do this - we should synchronize to not have "buf to hex string" written twice: https://github.com/northpolesec/santa/pull/241/files#diff-e8c43474389799d5b8375faaacae73c2404ce0b88f0d32b8094ff8f7cda3fe71R57

Reply:
I will review your PR now 😄